doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,377 @@
+# SONiC Storage Monitoring Daemon Design #
+### Rev 0.1 ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Ashwin Srinivasan  | Initial version                   |
+
+## 1. Overview
+
+This document is intended to provide a high-level design for a Storage monitoring daemon.
+
+Solid-State storage devices that use NAND-flash technology to store data offer the end user significant benefits compared to HDDs. Some advantages are reliability, reduced size, increased energy efficiency and improved IO speeds which translates to faster boot times, quicker computational capabilities and an improved system responsiveness overall. Like all devices, however, they experience performance degradation over time on account of a variety of factors such as overall disk writes, bad-blocks management, lack of free space, sub-optimal operational temperature and good-old wear-and-tear which speaks to the overall health of the disk. 
+
+The goal of the Storage Monitoring Daemon (storagemond) is to provide meaningful metrics for the aforementioned issues and enable streaming telemetry for these attributes so that preventative measures may be triggered in the eventuality of performance degradation.
```

Comment:
However,the document does not mention how to promptly notify administrators, it is preferable to print an alarm log when performance degradation occurs.

Reply:
Hi @philo-micas -- the idea here is that it is up to the end user to decide what 'degraded' performance looks like and take appropriate action. The daemon, for it's part, updates the statedb every loop.